### DISCUSSION TITLE
Programmatically get access to analog sensor

### DISCUSSION THREAD

**Discussion by Heizelmann** [arendst/Tasmota#18328] [Q&A]
I have a problem with analog sensors, e.g. a simple lumminance sensor, which I would like to access programmatically from my own driver module. I can not find that this sensor has a connection to TasmotaGlobal.

see also #18306

**Comment by arendst** [comment]
If you need access to other connected sensors I suggest you take a look at the prometheus driver (xsns_75_prometheus.ino) how they manage to pick up sensor data WITHOUT a total rewrite of sensor drivers.

**Reply by Heizelmann** [reply]
Sorry, but I can not see, how to pickup sensor data from analog sensor `xsns_02_analog.ino` I can not find an entry for this in `TasmotaGlobal`

**Reply by barbudor** [reply]
Theo didn't suggested you to look in the Analog driver but in the Prometheus driver which is a driver to allow the Prometheus database to pull sensor datas from a tasmota device. Which means that this driver is capable of getting any sensor data.
The trick is here : https://github.com/arendst/Tasmota/blob/b287d0509875347ec200440de405d99de82d60f6/tasmota/tasmota_xsns_sensor/xsns_75_prometheus.ino#L285-L346
